### PR TITLE
[VUFIND-1464] Add \Memcached support to Memcache session handler

### PR DESCRIPTION
**Pull request by demiankatz**
TODO:
- [x] Resolve [VUFIND-1464](https://vufind.org/jira/browse/VUFIND-1464) when merging.

### CODE REVIEW

**Comment by demiankatz** [comment]
@EreMaijala, I opened VUFIND-1464 a few months ago because you mentioned that PHP 7 and the \Memcache extension were incompatible. However, that does not seem to be the case -- I was able to successfully install both \Memcache and \Memcached with PHP 7 on my test server. In any case, while investigating this, I updated the Memcache session handler to support either client, which should help future-proof the code in case something really does break or get deprecated.

There's just one thing which I don't understand: one of the unit tests fails unless \Memcache::get() receives exactly three parameters, even though the documentation indicates that the method accepts two parameters, one of which is optional. I was able to make the test pass by sending some extra null parameters to the method, which should be harmless... but this is obviously an ugly hack. My best theory is that something in the \Memcache library is misreporting data to PHP's Reflection API, which is in turn making PHPUnit's mock library misbehave... but if that's the case, there's nothing I can do about it except the existing workaround. If you have any alternative theories or better solutions, though, I'd love to hear them!

**Comment by demiankatz** [comment]
Also, a side note: I notice that the Travis build did not run here. I am getting an error message about a negative credit balance, even though my account shows a positive balance. Perhaps Travis is having some technical difficulties right now, but if this does not resolve itself in a day or two, I will try to remember to contact support.

**Comment by demiankatz** [comment]
Actually, I take it back: I have a positive credit balance in my personal account, but the vufind-org account has run out. I'll see what I can do to resolve this with support!

**Comment by demiankatz** [comment]
Update: I've opened a ticket, but following a bit more research, I think the [prospects don't look good](https://news.ycombinator.com/item?id=25338983).

**Comment by EreMaijala** [comment]
@demiankatz Thanks, I'll take a look as time permits. I was indeed mistaken, and there is a current version of Memcache supporting PHP up to version 8.

File: module/VuFind/src/VuFind/Session/Memcache.php [hunk 1]
```diff
@@ -122,7 +122,12 @@ protected function connect(?Config $config, ?object $client): void
      */
     public function read($sessId)
     {
-        $value = $this->connection->get("vufind_sessions/{$sessId}");
+        // For some reason, Memcache tests fail if we do not pass exactly three
+        // parameters to the get method, even though this seems inconsistent with
+        // the documentation. This mechanism makes the tests pass, but may be worth
+        // revisiting in the future.
+        $extraParams = $this->connection instanceof \Memcache ? [null, null] : [];
+        $value = $this->connection->get("vufind_sessions/{$sessId}", ...$extraParams);
```
Comment (EreMaijala) [inline module/VuFind/src/VuFind/Session/Memcache.php]: This line seems to be too long.

**Review by EreMaijala — APPROVED** [review]

**Pull request merged**